src/config/mod.rs

```diff
@@ -116,14 +116,40 @@ pub struct Mouse {
 #[derive(Clone, Debug, Deserialize, PartialEq, Eq)]
 pub struct Url {
     // Program for opening links
-    #[serde(deserialize_with = "failure_default")]
+    #[serde(deserialize_with = "deserialize_launcher")]
     pub launcher: Option<CommandWrapper>,
 
     // Modifier used to open links
     #[serde(deserialize_with = "deserialize_modifiers")]
     pub modifiers: ModifiersState,
 }
 
+fn deserialize_launcher<'a, D>(deserializer: D) -> ::std::result::Result<Option<CommandWrapper>, D::Error>
+    where D: de::Deserializer<'a>
+{
+    // Deserialize to generic value
+    let val = match serde_yaml::Value::deserialize(deserializer) {
+        Ok(val) => val,
+        Err(err) => {
+            error!("Problem with config: {}; using default value", err);
```

Comment:
Is "using default value" the best message? What about "disabling URL luancher". I know this is a departure from convention, but it might be nice to start pointing users towards the failing configs in the message.

Reply:
Great point, this is just the generic error but with most other specialized deserializers we say exactly which value is used.